hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -2311,6 +2311,22 @@ public List<OzoneFileStatus> listStatus(String volumeName, String bucketName,
             allowPartialPrefixes);
   }
 
+  @Override
+  public List<OzoneFileStatusLight> listStatusLight(String volumeName, String bucketName, String keyName,
+      boolean recursive, String startKey, long numEntries) throws IOException {
+    OmKeyArgs keyArgs = prepareOmKeyArgs(volumeName, bucketName, keyName);
+    if (omVersion.compareTo(OzoneManagerVersion.LIGHTWEIGHT_LIST_STATUS) >= 0) {
+      return ozoneManagerClient.listStatusLight(keyArgs, recursive, startKey,
+          numEntries);
```

Comment:
`OzoneManagerVersion.LIGHTWEIGHT_LIST_STATUS` is an existing version, added in HDDS-11414.  The method being called is new in this PR:

https://github.com/apache/ozone/blob/bd264b5d415466743cf25825f9d4d75d1af26143/hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/protocol/OzoneManagerProtocol.java#L1000-L1001

So this is not compatible with versions of Ozone Manager between these commits (8ca33c718b and the one that will precede this on `master` when merged).

My suggestion:
- remove the new method from `OzoneManagerProtocol` and its implementations,
- call the existing method here:

```suggestion
      return ozoneManagerClient.listStatusLight(keyArgs, recursive, startKey,
          numEntries, false);
```

(It's not worth adding yet another `OzoneManagerVersion` constant, since the new method is a subset of the existing.)

Reply:
Why would it be incompatible? These are just interfaces right? If a server doesn't have the interface method it will anyways call the method with allowPartialPrefixes as false on the server side. There should be no incompatibilty in my opinion. Since one could have 2 different interfaces one server side and client side.

Reply:
Sorry, I was wrong, it works, because client never calls the new method on the server.  `OzoneManagerProtocolClientSideTranslatorPB` does the trick and calls the old method, somewhat like I suggested.

Still, I think removing the unused method would improve clarity.

Reply:
We have implemented the same thing even for listStatus API. It would be better to have some uniformity there.

Reply:
> We have implemented the same thing even for listStatus API. It would be better to have some uniformity there.

For `listStatus` methods, new method with additional parameter was introduced later:

```
2019-05-21 14:47:48 +0530  968)   List<OzoneFileStatus> listStatus(OmKeyArgs keyArgs, boolean recursive,
2019-05-21 14:47:48 +0530  969)       String startKey, long numEntries) throws IOException;

2022-06-03 08:41:29 +0530  984)   List<OzoneFileStatus> listStatus(OmKeyArgs keyArgs, boolean recursive,
2022-06-03 08:41:29 +0530  985)                                    String startKey, long numEntries,
2022-06-03 08:41:29 +0530  986)                                    boolean allowPartialPrefixes)
```

Since the existing `listStatusLight` covers both cases, overloading the method is not necessary now.